### PR TITLE
Standardize notebooks (somewhat)

### LINKED ISSUE #21: linting and formating
* Lint and format notebooks in `code/*` for standardization.
* Add a `pyproject.toml` with lint & format rules
* Add a `.pre-commit` so we can all stay healthy 💪

### LINKED ISSUE #25: add a standard license
add a standard license

### PR DESCRIPTION
**Pull request by jGaboardi**
This PR:
* resolves #21 
* resolves #25 

Will follow in order with:
* #24 
* #22 
* #23

### CODE REVIEW

**Comment by martinfleis** [comment]
James doing James-things! Love it :).

File: pyproject.toml
Comment (martinfleis) [inline pyproject.toml:1]: Given this is not a package, do we need `pyproject.toml` at all? If all we need is to configure ruff, I'd just use `ruff.toml` instead.

**Review by anastassiavybornova — APPROVED** [review]
Looks all good to me!

File: pyproject.toml
Comment (jGaboardi) [reply pyproject.toml:1]: I hadn't even thought about that. I will update.

**Comment by jGaboardi** [comment]
* I will make the corrections above and re-push.
* re: linting/formatting I think getting a GHA setup for this would be a but overboard, but I gan easily get that done if yall think it's a good idea. But if we simply agree to run `ruff` for `check` and `format` on commits through `pre-commit` then I think that's sufficient. @anastassiavybornova Do you know how to set up `pre-commit`?

**Comment by martinfleis** [comment]
pre-commit is preferable over action here

**Comment by anastassiavybornova** [comment]
hi! sorry just saw this. i've used pre-commit a few times before, will try to set it up and ping you on discord if something's not clear. so fine by me!

**Review by martinfleis — APPROVED** [review]

**Pull request merged**